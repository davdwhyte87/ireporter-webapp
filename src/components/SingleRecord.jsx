import React from 'react'
import { connect } from 'react-redux'
import propTypes from 'prop-types'
import { toast } from 'react-toastify'
import { getRecord } from '../actions/recordActions'
import { Link, withRouter } from 'react-router-dom';
import Alert from './Alert'
import RecordView from './RecordView'

class SingleRecord extends React.Component {
  constructor(props) {
    super(props)
  }
  componentDidMount() {
    const id = this.props.match.params.id
    this.props.getRecord(id)
  }
  render() {
    const user = JSON.parse(localStorage.getItem('user'));
    console.log(this.props)
    if (this.props.loading) {
      return (
        <section className="records" id="records">
          <div className="container-center">Loading.....</div>
        </section>
        )
    } else {
      return (
        <section className="records" id="records">
          {this.props.errors? (
            <section>
             <Alert message={this.props.errors}  />
            </section>
          ) :
           ('')
          }
          {this.props.record ?
            (
              <RecordView user={user} record={this.props.record} />
            )
            : '' }
        </section>
      )
    }
  }
}

SingleRecord.propTypes = {
  getRecord: propTypes.func.isRequired
}

SingleRecord.defaultValues = {
  loading: false
}
const mapStateToProps = state => {
  return {
    errors: state.recordsReducer.errors,
    loading: state.recordsReducer.loading,
    success: state.recordsReducer.success,
    record: state.recordsReducer.record
  }
}

export default connect(() => mapStateToProps, { getRecord })(withRouter(SingleRecord))