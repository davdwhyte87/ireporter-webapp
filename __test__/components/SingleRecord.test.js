import React from 'react';
import { mount } from 'enzyme';
import SingleRecord from '../../src/components/SingleRecord';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

describe('Single record view test', () => {
  const mockStore = configureStore([thunk]);
  const store = mockStore({
    recordsReducer: {
      success: false,
      records: [],
      record: null
    }
  });
  it('should render without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
      <BrowserRouter>
       <SingleRecord />
      </BrowserRouter>
    </Provider>
    );
    expect(wrapper.find('.records').exists()).toBe(true);
  });

  it('should render without crashing', () => {
    const storeLoading = mockStore({
      recordsReducer: {
        success: false,
        loading: true,
        records: [],
        record: null
      }
    });
    const wrapper = mount(
      <Provider store={storeLoading}>
      <BrowserRouter>
       <SingleRecord />
      </BrowserRouter>
    </Provider>
    );
    expect(wrapper.find('.records').exists()).toBe(true);
  });

  it('should match snapshot', () => {
    const wrapper = mount(
      <Provider store={store}>
      <BrowserRouter>
        <SingleRecord />
      </BrowserRouter>
    </Provider>
    );
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  })
});