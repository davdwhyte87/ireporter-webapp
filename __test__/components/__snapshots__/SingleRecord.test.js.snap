// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single record view test should match snapshot 1`] = `
<section
  className="records"
  id="records"
>
  
  
</section>
`;
